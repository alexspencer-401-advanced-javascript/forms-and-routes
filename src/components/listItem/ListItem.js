import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const ListItem = ({ _id, name, image }) => {
  return (
    <li key={_id}>
      <figure>
        <img src={image} alt={name}></img>
        <Link to={`/character/${_id}`}>
          <figcaption>{name}</figcaption>
        </Link>
      </figure>
    </li>
  );
};

ListItem.propTypes = {
  _id: PropTypes.string,
  name: PropTypes.string,
  image: PropTypes.string,
};

export default ListItem;
