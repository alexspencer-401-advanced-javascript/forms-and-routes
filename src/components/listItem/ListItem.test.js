import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';

describe('ListItem component', () => {
  it('renders ListItem', () => {
    const wrapper = shallow(<ListItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
