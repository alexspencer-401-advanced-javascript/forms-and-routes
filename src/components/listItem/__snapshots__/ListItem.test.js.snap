// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders ListItem 1`] = `
<li>
  <figure>
    <img />
    <Link
      to="/character/undefined"
    >
      <figcaption />
    </Link>
  </figure>
</li>
`;
